lib/github_changelog_generator/generator/entry.rb

```diff
@@ -0,0 +1,309 @@
+require "github_changelog_generator/generator/section"
+
+module GitHubChangelogGenerator
+  class Entry
+    attr_reader :content
+
+    def initialize(options = {})
+      @content = ""
+      @options = options
+    end
+
+    # Generates log entry with header and body
+    #
+    # @param [Array] pull_requests List or PR's in new section
+    # @param [Array] issues List of issues in new section
+    # @param [String] newer_tag Name of the newer tag. Could be nil for `Unreleased` section
+    # @param [Hash, nil] older_tag Older tag, used for the links. Could be nil for last tag.
+    # @return [String] Ready and parsed section
+    def create_entry_for_tag(pull_requests, issues, newer_tag, older_tag = nil)
+      newer_tag_link, newer_tag_name, newer_tag_time = detect_link_tag_time(newer_tag)
+
+      github_site = @options[:github_site] || "https://github.com"
+      project_url = "#{github_site}/#{@options[:user]}/#{@options[:project]}"
+
+      # If the older tag is nil, go back in time from the latest tag and find
+      # the SHA for the first commit.
+      older_tag_name =
+        if older_tag.nil?
+          @fetcher.commits_before(newer_tag_time).last["sha"]
+        else
+          older_tag["name"]
+        end
+
+      set_sections_and_maps
+
+      @content = generate_header(newer_tag_name, newer_tag_link, newer_tag_time, older_tag_name, project_url)
+
+      @content += generate_body(pull_requests, issues)
+
+      @content
+    end
+
```

Comment:
All following methods should be private, and `create_entry_for_tag` should be tested.